### PR TITLE
Add HPO sweep support to other Methods than BaselineMethod

### LINKED ISSUE #98: Add HPO Sweep functionality to other Methods than BaslineMethod
At the moment, the only Method with the 'built-in' HPO tuning with Orion is the BaselineMethod.

Would be nice if instead we moved that logic to the Method class, and made it possible for Methods to override some method stubs to enable them to be tuned.

### PR DESCRIPTION
**Pull request by lebrice**
Fixes #98  
See the commit messages for more info, but this basically:

- Adds optional support for running HPO sweeps on all Methods, as long as they provide an implementation for the following two functions:
    * `Method.get_search_space(self, setting: Setting) -> Dict` : Returns the search space to be used by Orion
    * `Method.adapt_to_new_hparams(self, new_hparams: Dict) -> None`: Adapts the Method to use the new hyper-parameter values.
    
    All Methods that inherit from `BaselineMethod` or from `StableBaselines3Method` will already have these methods implemented. 

- Adds hyper-parameters and their priors to the following Methods:
    * [X] EWCMethod
    * [X] DQN
    * [X] PPO
    * [X] A2C
    * [ ] (All other SB3 methods only have a prior on the learning rate as of right now, i.e. only the learning rate is tunable)

TODOs:
- [ ] Should we add the necessary methods to the examples? Or should we keep those simple for now?

### CODE REVIEW

**Comment by lebrice** [comment]
I'll close this out earlier, rather than later, since things are getting a bit messy.

**Comment by lebrice** [comment]
Fixes #103 , as far as I can tell, since it also removes the core dependency on `atari_py`.
